types/services.d.ts

```diff
@@ -213,6 +209,205 @@ declare namespace Realm {
              */
             type AggregatePipelineStage = object;
 
+            /**
+             * An operation performed on a document.
+             */
+            enum OperationType {
```

Comment:
No - I think you're right here. I just copied the enum from the legacy SDK.

Reply:
One advantage of an `enum` is that it enables us to document the semantics of the various values.

Reply:
The rename and drop operations are not obvious in my opinion.
I don't think those comments will find its way into an editor or a jsdoc generated site.

Reply:
Lets make this a `const enum` so that it can only be used in ways that completely compile away. We don't want people to be doing operations that assume that there is some `OperationType` object.

Reply:
Actually, it looks like the semantics for enums aren't quite right ([Playground Link](https://www.typescriptlang.org/play/#code/MYewdgzgLgBApmArgWxgDRgbxgQxgXhgCIciYBfAKABs5YAPALnQONIG4YB6LmAFQCeABzgwA5CSJiYASwgwwIWDggQZAczA4ARrRhQQ+4aLFoxAOgAUAJgDM16wEogA)). Maybe we should go back to union type of string literals and put the documentation for each kind in the docstring for the type (or just refer people to the mongo docs)?